common-lib/src/cmr/common/util.clj

```diff
@@ -764,6 +774,24 @@
   [m k]
   (or (get m k) (lazy-get m k)))
 
+(defn delazy-value
+  "Remove a lazy value and replace it with the actual value"
+  [map key]
+  (if-let [actual (lazy-get map key)]
+    (-> map
+        (assoc key actual)
+        (dissoc (key->delay-name key)))
+    map))
+
+(defn delazy-all
+  "In a map, Remove all the lazy values and replace them with actual values so
```

Comment:
can you use (doall (function call ...) instead of this function?  (doall changes a lazy function to a non lazy function.)

Reply:
is lazy the same as delay? I'll have to do a test.

Reply:
Lazy will only pull the data when necessary. so if you have a list that you are iterating over but do nothing with it, the iteration won't happen. Of if you have a db connection and you have queried a table, it won't actually load the results until you start using it. It could be delayed, but it doesn't mean the exact same thing.